src/xdsl/dialects/linalg.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+from xdsl.ir import *
+from xdsl.irdl import *
+from xdsl.util import *
+from xdsl.dialects.memref import *
+from xdsl.dialects.builtin import IntegerType, Float32Type, IntegerAttr, FlatSymbolRefAttr
+from xdsl.parser import Parser
+
+
+@dataclass
+class Linalg:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_op(Generic)
+        self.ctx.register_op(Yield)
+        self.ctx.register_op(Fill)
+        self.ctx.register_op(Index)
+
+
+@irdl_op_definition
+class Generic(Operation):
+    name: str = "linalg.generic"
+    inputs = VarOperandDef(AnyAttr)
+    outputs = VarOperandDef(AnyOf([MemRefType, TensorType]))
+    indexing_maps = AttributeDef(ArrayAttr)
+    iterator_types = AttributeDef(ArrayAttr)
+    doc = AttributeDef(StringAttr)
+    library_call = AttributeDef(StringAttr)
+
+    region = RegionDef()
+    output = VarResultDef(TensorType)
+
+    irdl_options = [AttrSizedOperandSegments()]
+
+    @staticmethod
+    def parse(parser: Parser) -> Generic:
+
+        attributes = parser.parse_op_attributes()
+        parser.parse_string("ins")
```

Comment:
The problem I see in your filecheck test is a bit complex: 

@math-fehr It seems that `parse_string` will never complain if the passed string is not present, it simply does nothing and returns `True`. 
Due to the above issue this line is effectively a no-op and the parser continues with parsing the operands. As the input in the test does not satisfy the structure here, it will fail on the second `parse_operands`. 

Note that this custom parser will always be executed, even if the input is in a form that the normal parser could handle.

Reply:
I'm extremely ashamed of this function, I don't know how I manage to write something that bad!
Yes, I'll fix this!

Reply:
All good, don't forget that most of the parser was written in a very short time. Furthermore, we didn't have too many tests at this point in time.